lib/basket.rb

```diff
@@ -0,0 +1,54 @@
+require_relative 'takeaway'
+
+class Basket
+
+  attr_reader :basket
+
+  def initialize(menu)
+    @basket = []
+    @menu = menu
+  end
+
+  def view
+    "Your basket total is #{basket_total} and it contains:\n#{basket_list}"
+  end
+
+  def add(item, price = @menu.read[item], quantity = get_quantity(item))
+    @basket << hash(item, price, quantity)
+  end
+
+  private
```

Comment:
you're getting low test coverage on this class because it's saying that some of your private methods aren't covered. although it's right not to test private method directly you should still be able to test their behaviour indirectly by writing appropriate tests for the public methods where they're used... i think.... for instance respec is saying that your private methos `basket_total`, `basket_list` and `compose_line` are covered, even though you haven't tested them directly. so i think you should be able to get coverage for the other ones. maybe bring this up with katerina...